src/coreclr/vm/threads.cpp

```diff
@@ -376,6 +376,7 @@ void SetThread(Thread* t)
     LIMITED_METHOD_CONTRACT
 
     gCurrentThreadInfo.m_pThread = t;
+    EnsureTlsDestructionMonitor(t);
```

Comment:
With this change, the destructor would need to fetch the thread via `GetThreadNULLOk()` to make sure we are not touching a thread that is dead (not sure if what we do is safe in such case).

That might be a good idea though if we want a minimal change for 6.0 porting purposes.

Reply:
`~TlsDestructionMonitor` calls `DetachThread`. `DetachThread` calls `SetThread(NULL)`. I think it would be a good idea to avoid calling `EnsureTlsDestructionMonitor` from `SetThread(NULL)` on this path to avoid attempt to re-register object that is being destructed.

Reply:
Yes, I just mean we should not rely on storing thread in m_thread and should fetch it via `GetThreadNULLOk()`. 

If I use `m_thread` I get a lot of asserts when switching mode when there is no current thread any more.